### DISCUSSION TITLE
[SF] HTTP client settings for Service Fabric?

### DISCUSSION THREAD

**Discussion by linmeng08** [dotnet/yarp#892] [Q&A] [answered]
Hi, how can I specify HTTP client settings for YARP proxy service running in Service Fabric cluster? For example:

1. Use a client certificate to connect to the destination endpoints
2. Customize HTTPS certificate validation for destination endpoint - e.g. validation certificate chain but ignore host name not matching error, as the destination endpoints are typically private IP address of the nodes like https://10.0.0.5:20012/269aec54-396c-4658-9790-4a7e6760a225/132621760379835038.

I can see the [Proxy HTTP Client Configuration](https://microsoft.github.io/reverse-proxy/articles/proxyhttpclientconfig.html) section has documentation on how to configure those settings for each cluster, but in case of Service Fabric the clusters are auto-discovered and not configured in the configuration files. In the [SF service enlistment section](https://microsoft.github.io/reverse-proxy/articles/service-fabric-int.html#sf-service-enlistment) there is option for specifying HTTP request settings (timeout, etc) for the backend services but no client certificates and other settings.

Is using a [Custom IProxyHttpClientFactory](https://microsoft.github.io/reverse-proxy/articles/proxyhttpclientconfig.html#custom-iproxyhttpclientfactory) the suggested way for this?

**Comment by Tratcher** [comment, marked as answer]
Here's the code that maps those options (new in preview 10, #787):
https://github.com/microsoft/reverse-proxy/blob/c3c0407b613286734cfb9d37102a933840a7f2d1/src/ReverseProxy.ServiceFabric/ServiceDiscovery/Util/LabelsParser.cs#L310-L326

Client certificates aren't supported in this model, you'll need to use a custom factory for those. https://github.com/microsoft/reverse-proxy/pull/869 should make that easier.

**Reply by linmeng08** [reply]
Cool I will try it out, thanks @Tratcher !
